RIOT-OS/RIOT
# boards/samr21-xpro: Model features in Kconfig

### Contribution description
This PR models the `FEATURES_PROVIDED` as Kconfig symbols. The idea is, similarly to the current `Makefile.features`, to select the provided features at every level (i.e. board, CPU model, CPU series, etc.).

~~In `cpu/Kconfig` and `board/Kconfig` I declared common CPU and board symbols, which take their value depending on the selected CPU model and board respectively. Here I proposed to have `BOARD`, `CPU_MODEL`, `CPU_SERIES`, `CPU_FAMILY` and `CPU_ARCH` (more specific to less specific), plus `CPU` which matches the current `CPU` variable. We should evaluate if that cover our use cases properly.~~

~~The selection of symbols starts with the board and goes like:~~
```
BOARD_<board> --[select]--> CPU_MODEL_<model> --[select]--> CPU_SERIES_<series> --[select]--> CPU_FAMILY_<family> --[select]--> CPU_ARCH_<arch>
```
~~Although this chain of symbols tries to mirror the common symbols, nothing prevents use from using other intermediate symbols (e.g. here I added `CPU_ARCH_CORTEX_M0PLUS` which selects `CPU_ARCH_CORTEX_M`).~~

Done in #13715.

#### On features
~~I added `cpu/Kconfig.features` for the declaration of *common* feature-like symbols. For features specific to a CPU, board or architecture we can declare them directly on the corresponding Kconfig file. Here I used the `HAS_` prefix to indicate features, mostly because it is customary in Kconfig files across projects, but we could use something else.~~

Done in #14151.

In the future, we could also indicate via this mechanism that, for instance, the `CPU_MODEL_SAMD21G18A` presents an at86rf233 radio.

### Testing procedure
I added the `debug-features-provided-kconfig` target in `dependencies_debug.inc.mk`, which should print the same list as `info-features-provided`, and will allow us to test the migration process. To test it, make sure you set `SHOULD_RUN_KCONFIG=1`.

Also, one can run `make menuconfig` and either enable the 'show-all mode' (by pressing `a`) or search for a particular feature symbol (by presing `/`), like `HAS_PERIPH_GPIO`. Once the symbol is selected, one can see more information on the symbol (by pressing `?`), like the description of the feature and who selects (provides) it (the board, the cpu model, etc.).

**Edit**: A test `tests/kconfig_features` has been added to check that boards provide the same features via Kconfig and via `Makefile.features`. This is useful during migration process to ensure sync.

### Issues/PRs references
~~Depends on #14151~~
